pkg/gke/components/GKENodePool.vue

```diff
@@ -0,0 +1,488 @@
+
+    name: {
+      type:    String,
+      default: ''
+    },
+
+    initialNodeCount: {
+      type:    [String, Number],
+      default: ''
+    },
+
+    maxPodsConstraint: {
+      type:    [String, Number],
+      default: ''
+    },
+
+    autoscaling: {
+      type:    Boolean,
+      default: false
+    },
+
+    autoRepair: {
+      type:    Boolean,
+      default: false
+    },
+
+    autoUpgrade: {
+      type:    Boolean,
+      default: false
+    },
+
+    minNodeCount: {
+      type:    [String, Number],
+      default: ''
+    },
+
+    maxNodeCount: {
+      type:    [String, Number],
+      default: ''
+    },
+
+    oauthScopes: {
+      type:    Array as PropType<string[]>,
+      default: () => []
+    },
+
+    rules: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    }
+  },
+
+  data() {
+    return { upgradeKubernetesVersion: false, initialVersion: this.version };
+  },
+
+  watch: {
+    upgradeKubernetesVersion(neu) {
+      if (neu) {
+        this.$emit('update:version', this.clusterKubernetesVersion);
+      } else {
+        this.$emit('update:version', this.initialVersion);
+      }
+    },
+
+    clusterKubernetesVersion: {
+      handler(neu) {
+        if (neu && this.mode === _CREATE) {
+          this.$emit('update:version', neu);
+        }
+      },
+      immediate: true
+    },
+
+    autoscaling(neu) {
+      if (!neu) {
+        this.$emit('update:minNodeCount', null);
+        this.$emit('update:maxNodeCount', null);
+      } else {
+        this.$emit('update:minNodeCount', '1');
+        this.$emit('update:maxNodeCount', '3');
+      }
+    }
+  },
+
+  computed: {
+    ...mapGetters({ t: 'i18n/withFallback' }),
+
+    _VIEW() {
+      return _VIEW;
+    },
+
+    // when initially created nodepools are configured to use the same k8s version as the control plane (clusterKubernetesVersion)
+    // on edit, if the cp version is updated, the user is given the option to update each node pool as well
+    upgradeAvailable() {
+      if (this.mode === _CREATE) {
+        return false;
+      }
+
+      return this.initialVersion !== this.clusterKubernetesVersion;
+    },
+
+    imageTypeOptions() {
+      return imageTypes.map((type) => {
+        return {
+          value: type,
+          label: this.t(`gke.imageType.${ type }`, null, type)
+        };
+      });
+    },
+
+    diskTypeOptions() {
+      return [
+        {
+          label: this.t('gke.diskType.standard'),
+          value: 'pd-standard',
+        },
+        {
+          label: this.t('gke.diskType.ssd'),
+          value: 'pd-ssd',
+        }
+      ];
+    },
+
+    selectedImageType: {
+      get() {
+        return this.imageTypeOptions.find((opt) => opt.value === this.imageType) || { value: this.imageType, label: this.t(`gke.imageType.${ this.imageType }`, null, this.imageType) };
+      },
+      set(neu: {label: string, kind?: string, value?: string, disabled?: boolean, [key: string]: any}) {
```

Comment:
`label: string, kind?: string, value?: string, disabled?: boolean, [key: string]: any` is used 6 times, would be worth being a `type` somewhere

Reply:
I added types for a handful of labeledselect option lists here and in networking.